theano/scan_module/scan_checkpoints.py

```diff
@@ -96,16 +101,29 @@ def scan_checkpoints(fn, sequences=[], outputs_info=None, non_sequences=[],
     if n_steps is None:
         n_steps = sequences[0].shape[0]
 
-    # Compute the number of steps of the inner and of the outer scan
-    o_n_steps = theano.tensor.cast(n_steps / save_every_N, 'int64')
-    i_n_steps = save_every_N
+    # Compute the number of steps of the outer scan
+    o_n_steps = theano.tensor.cast(theano.tensor.ceil(n_steps / save_every_N),
+                                   'int64')
+    i_n_steps = save_every_N * theano.tensor.ones((o_n_steps,), 'int64')
+    mod = n_steps % save_every_N
+    sign = theano.tensor.sgn(mod)
```

Comment:
We do not support negative steps in scan anymore. So go back to the simplest earlisest version. Where did you saw in the doc that we support negative steps? We should remove that from the doc.